File: src/codility/java/counting/FrogRiverOne.java
package codility.java.counting;

import java.util.HashSet;

public class FrogRiverOne {
    public int solution(int X, int[] A) {
        // write your code in Java SE 8
        HashSet<Integer> set = new HashSet<>();

        int ans = -1;

        for (int i = 0; i < A.length; i++) {
            int a = A[i];
            if (a <= X) {
                set.add(a);
            }

            if (set.size() == X) {
                ans = i;
                break;
            }
        }
        return ans;
    }
}

File: src/codility/kotlin/FrogRiverOne.kt
package codility.kotlin

class FrogRiverOne {

    fun solution(X: Int, A: IntArray): Int {
        // write your code in Kotlin

        val set = mutableSetOf<Int>()

        for (i in A.indices) {
            set.add(A[i])
            if (set.size == X) {
                return i
            }
        }
        return -1

    }

}